connectors/aws-cert/src/main/java/com/vmware/connectors/aws/cert/AwsCertController.java

```diff
@@ -50,26 +54,37 @@
 
     private static final String ROUTING_PREFIX = "x-routing-prefix";
 
+    private final static String METADATA_PATH = "/discovery/metadata.json";
+
     private static final String APPROVE_PATH = "/api/v1/approve";
 
     private final String certificateApprovalHost;
     private final String certificateApprovalPath;
+    private final String metadata;
     private final WebClient rest;
     private final CardTextAccessor cardTextAccessor;
 
     @Autowired
     public AwsCertController(
             @Value("${aws.certificate.connector.approval.host}") String certificateApprovalHost,
             @Value("${aws.certificate.connector.approval.path}") String certificateApprovalPath,
+            @Value("classpath:static/discovery/metadata.json") Resource metadataJsonResource,
             WebClient rest,
             CardTextAccessor cardTextAccessor
-    ) {
+    ) throws IOException {
         this.certificateApprovalHost = certificateApprovalHost.toLowerCase(Locale.US);
         this.certificateApprovalPath = certificateApprovalPath;
+        this.metadata = IOUtils.toString(metadataJsonResource.getInputStream(), Charset.defaultCharset());
         this.rest = rest;
         this.cardTextAccessor = cardTextAccessor;
     }
 
+    @GetMapping(path = METADATA_PATH)
+    public ResponseEntity<String> getMetadata(HttpServletRequest request) {
+        String requestUrl = request.getRequestURL().toString();
+        return ResponseEntity.ok(this.metadata.replace("CONNECTOR_HOST", requestUrl.split(METADATA_PATH)[0]));
```

Comment:
If you're going to do a string replace on the metadata.json contents, use something less likely to come up in a real url and more obvious that it will be replaced: `@@CONNECTOR_HOST@@` or `${CONNECTOR_HOST}` or something

Reply:
Yes, also this might help quickly identify there is something to be replaced, while reading.